### DISCUSSION TITLE
Unable to understand how the decompress quaternion from 'stateEstimateZ.quat'

### DISCUSSION THREAD

**Discussion by ReanFernandes** [bitcraze/discussions#560] [Q&A]
I am using the python api to get the compressed information from my drone, which include the compressed quaternion obtained from 'stateEstimateZ.quat'. The page containing the logging variable information however, has incomplete documentation, with a link that i think was not added ( https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/logs/#stateestimatez ).
While checking the code given at (https://github.com/bitcraze/crazyflie-firmware-experimental/blob/master/src/modules/interface/quatcompress.h),  I am unable to grasp what the order of the quaternions return by the decompress function will be. I assume it should be of the form [qw, qx, qy, qz], but i would like it if you could clarify this for me. 
Also If there is something that is missing in the logging variable page, I request you to update it so that we can use it without this confusion. 
I hope my question was clear, and if not I will be glad to give you more information. 
Thank you for your time! :)

**Comment by knmcguire** [comment]
Hi! This log is mostly used in the Crazyswarm(2) framework, so there is only an c++ example here:
https://github.com/IMRCLab/crazyswarm2/blob/ec8f21823cf52e58810c81b051434692ea4f516d/crazyflie/src/crazyflie_server.cpp#L556-L588 which is located in @whoenig crazyflie_cpp library:
https://github.com/whoenig/crazyflie_cpp/blob/016ffd288f0959bb9b84e7afc9d47200af9e0caf/src/crtp.cpp#L11

Technically... we should have this function in our [cflib](https://github.com/bitcraze/crazyflie-lib-python) as well.. but it hasn't been implemented, so I hope that the c++ library could give some pointers.

**Reply by knmcguire** [reply]
Check out this issue of the cflib feature request: https://github.com/bitcraze/crazyflie-lib-python/issues/378

**Reply by ReanFernandes** [reply]
Thank you for the quick reply, I'll test this out and get back on whether it is a fix. :)

**Reply by ReanFernandes** [reply]
Hello!
After implementing the function you provided, I cannot seem to figure out the order in which the quaternion vector is arranged. Since qw is the highest among all the four, I was able to find that, but as to the rest of the quaternions, I have no idea. Please forgive my ignorance, if its an obvious answer, but could you clarify the order of the quaternions after the vector has been decompressed?

**Reply by whoenig** [reply]
It says it in the comment: `output q = [x,y,z,w]`. Note that this is different from [rowan](rowan.readthedocs.io/).
